parquet/src/main/java/org/apache/iceberg/parquet/PruneColumns.java

```diff
@@ -71,11 +73,14 @@ public Type struct(GroupType struct, List<Type> fields) {
     for (int i = 0; i < fields.size(); i += 1) {
       Type originalField = struct.getType(i);
       Type field = fields.get(i);
-      if (selectedIds.contains(getId(originalField))) {
-        filteredFields.add(originalField);
-      } else if (field != null) {
-        filteredFields.add(originalField);
-        hasChange = true;
+      Integer fieldId = getId(originalField);
+      if (fieldId != null) {
```

Comment:
Same logic as above applies here.

Reply:
And in the rest of the updates for this file.

Reply:
Done.